jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpFields.java

```diff
@@ -1204,6 +1204,167 @@ else if (ensured > 0)
             // return a merged header with missing ensured values added
             return new HttpField(ensure.getHeader(), ensure.getName(), v.toString());
         }
+
+        /**
+         * A wrapper of {@link HttpFields}.
+         */
+        class Wrapper implements Mutable
+        {
+            private final Mutable _fields;
+
+            public Wrapper(Mutable fields)
+            {
+                _fields = fields;
+            }
+
+            /**
+             * Called when a field is added (including as part of a put).
+             * @param field The field being added.
+             * @return The field to add, or null if the add is to be ignored.
+             */
+            public HttpField onAddField(HttpField field)
```

Comment:
The `onAddField()` method is being called twice for each add.
Once in `Wrapper.add()` and the other in `ListIterator` implementation.